client.go

```diff
@@ -109,30 +113,40 @@ type Client interface {
 
```

Comment:
Do you think it's also worth updating the functions we currently have commented out?
There are a few others that we might also want to update as well in that case